sqlmesh/core/loader.py

```diff
@@ -42,6 +45,24 @@
 logger = logging.getLogger(__name__)
 
 
+_cache = None
+
+
+def _initializer(optimized_query_cache: OptimizedQueryCache) -> None:
+    global _cache
+    _cache = optimized_query_cache
+
+
+def _load_cache(model: Model) -> t.Tuple[str, t.Optional[str]]:
+    assert _cache
+    if isinstance(model, SqlModel):
+        entry_name = _cache._entry_name(model)
```

Comment:
Why do you need this? Why not just call put without an argument and let it compute the entry name?

Reply:
Or can we just pass cache as an argument instead?

Reply:
because computing the hash entry isn't free, it calls gen which is a full traversal of ast

Reply:
Then I suggest making the `_entry_name` method public.